python/lsst/ts/m2gui/display/figure_constant.py

```diff
@@ -123,45 +147,250 @@ def _set_default_series(self, range_x_min, range_x_max, num_points):
+
+        values = list()
+        for idx in range(num_series):
+            value_min, value_max = self._get_range_points(idx)
+            values.append(value_min)
+            values.append(value_max)
+
+        self._value_y_min = min(values)
+        self._value_y_max = max(values)
+        self._update_range_axis_y(threshold=1)
+
+    def _get_range_points(self, idx):
+        """Get the range of points in a specific series.
+
+        Parameters
+        ----------
+        idx : `int`
+            Index of the series.
+
+        Returns
+        -------
+        `float`
+            Minimum value.
+        `float`
+            Maximum value.
+        """
+
+        points = self.get_points(idx)
+
+        values = list()
+        for point in points:
+            values.append(point.y())
+
+        return min(values), max(values)
+
+    def append_data(self, value, idx=0):
+        """Append the data. This is only useful if the realtime data is
+        applied.
+
+        Parameters
+        ----------
+        value : `float`
+            New coming data.
+        idx : `int`, optional
+            Index of the series. (the default is 0)
 
-        series = self.chart().series()[0]
-        count = series.count()
-        if num_x > count:
-            raise ValueError(f"Input x number(={num_x}) > available points (={count}).")
+        Raises
+        ------
+        `RuntimeError`
+            If not for the realtime data.
+        """
+
+        if not self._is_realtime:
+            raise RuntimeError("Can only be called for the realtime data.")
 
-        points = [QPointF(x, y) for (x, y) in zip(list_x, list_y)]
+        points = self.get_points(idx)
+        self._append_point(points, value)
+
+        # Replace the points in series
+        series = self.get_series(idx)
         series.replace(points)
 
-        self.axis_y.setRange(min(list_y) - self.OFFSET_Y, max(list_y) + self.OFFSET_Y)
+        # Update the range of axes to show the new data
+        if self.axis_x.max() < points[-1].x():
+            self.axis_x.setRange(points[0].x(), points[-1].x())
+
+        # Check the range of y-axis needs to be updated or not
+        self._value_y_max = max(self._value_y_max, value)
+        self._value_y_min = min(self._value_y_min, value)
+        self._update_range_axis_y()
+
+    def _append_point(self, points, value):
+        """Append the point.
+
+        Parameters
+        ----------
+        points : `list [PySide2.QtCore.QPointF]`
+            Points.
+        value : `float`
+            New coming data.
+        """
+
+        length_points = len(points)
+
+        # No data yet, begin from 0
+        if length_points == 0:
+            points.append(QPointF(0, value))
+
+        # The holded amount of data is less than the allowed maximum. Add the
+        # new point.
+        elif length_points < self._num_points:
+            points.append(QPointF(points[-1].x() + 1, value))
```

Comment:
Please make up your mind. Either you are creating LineSerie where X are real numbers, or you are creating a series where X are integers. You cannot have both. Initialization with np.linspace suggest you are using real numbers, yet there you append integers. Also, those two appends can be simplified to:
```
    if length_points < self._num_points:
        points.append(QPointF(length_points - 1, value))
```
which does the same (assuming X are integers).

Reply:
The functions of `self.append_data()` and `self.update_data()` can not be called at the same time because of the checking of `self._is_realtime`. For the `self.append_data()`, the x axis will be only the integer as a mimic of LabVIEW default chart behavior. Thanks!